tsgen/__main__.py

```diff
@@ -132,12 +134,16 @@ def generate_on_car_telemetry(self):
                             which casts the pointer to the bytes to unpack into to the actual
                             type and calls the unpacking function.
                  */
-                #define IMPL_HANDLER(T, f) \\
-                    static void T##_handler(uint8_t* dst, const uint8_t* src, size_t length) \\
+                #define IMPL_HANDLER(T, p, u) \\
+                    static void T##_handler_pack(uint8_t* dst, const uint8_t* src, size_t length) \\
                     { \\
-                        struct T* output = (struct T*) dst; \\
```

Comment:
I would rather not remove this cast, just to make the intent explicit even if functionally it does nothing. The compiler should also give a warning if it's left out?

Reply:
The purpose of this cast is to convert the `uint8_t` pointer of the unpacking destination to a pointer to the `struct` type of the packing function. Without this `uint8_t*` is being passed as the argument to a function where the parameter type is `T*`.

Reply:
There are no compiler warnings about it. 
I understand your intention here, but still don't understand how it works specifically here. It cast pointer `dst` to pointer  `output`. Where is `output` used later in function? If you're assigning `output`, is it actually converting dst?

`f( (struct T*) dst, src, length); \\` how about casting in that way?

Reply:
Yeah you're right that `output` is never used later in the function, casting it in the way you've suggested is fine. I think the cast should definitely be there in whatever form.

Strange that you don't get any compiler warnings, is this with the on-car telemetry build system?

Reply:
I can't remember the name of the warning, but I feel like it should be "incompatible pointer type" or something.